jetty-websocket/websocket-core/src/main/java/org/eclipse/jetty/websocket/core/client/HttpClientProvider.java

```diff
@@ -16,22 +16,37 @@
 //  ========================================================================
 //
 
-package org.eclipse.jetty.websocket.client.impl;
+package org.eclipse.jetty.websocket.core.client;
 
 import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.util.log.Log;
 import org.eclipse.jetty.util.ssl.SslContextFactory;
 import org.eclipse.jetty.util.thread.QueuedThreadPool;
 
-class DefaultHttpClientProvider
+public interface HttpClientProvider
 {
-    public static HttpClient newHttpClient()
+    static HttpClient get()
     {
-        SslContextFactory sslContextFactory = new SslContextFactory();
-        HttpClient client = new HttpClient(sslContextFactory);
+        try
+        {
+            HttpClientProvider xmlProvider = new XmlHttpClientProvider();
+            HttpClient client = xmlProvider.newHttpClient();
+            if (client != null)
+                return client;
+        }
+        catch (Throwable ignore)
+        {
+            Log.getLogger(HttpClientProvider.class).ignore(ignore);
+        }
+
+        return new HttpClientProvider(){}.newHttpClient();
```

Comment:
Let's use one more feature of interfaces.

They can now have private static methods, so:

1. Move the implementation of `newHttpClient()` to a private static method called `newDefaultHttpClient()`.
2. Here, rather than creating a new anonymous implementation, just call the private static method.